pkg/controller/cloudwatchmetrics/cloudwatchmetrics_controller.go

```diff
@@ -0,0 +1,111 @@
+package cloudwatchmetrics
+
+import (
+	"context"
+	"time"
+
+	integreatlyv1alpha1 "github.com/integr8ly/cloud-resource-operator/pkg/apis/integreatly/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_cloudwatchmetrics")
+
+// Set the reconcile duration for this controller.
+// Currently it will be called once every 5 minutes
+const watchDuration = 5
+
+// Add creates a new CloudwatchMetrics Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileCloudwatchMetrics{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("cloudwatchmetrics-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Send a generic event to the channel to kick
+	// off the first reconcile
+	events := make(chan event.GenericEvent)
+	go func() {
+		events <- event.GenericEvent{
```

Comment:
Could we document at the top of this file the approach we're taking here.

- That this controller is for reconciling metrics for every resource (currently only redis and postgres)
- That this controller takes a sync the world approach
- That this controller will reconcile every set period of time

To make sure it's clear for future readers of the code

Reply:
Sure